ibis/backends/pyspark/tests/test_basic.py

```diff
@@ -149,72 +96,26 @@ def test_join(con):
         ),
     ],
 )
-def test_filter(con, filter_fn, expected_fn):
-    table = con.table("basic_table")
-
-    result = filter_fn(table).compile()
-
-    df = table.compile()
-    expected = expected_fn(df)
-
-    tm.assert_frame_equal(result.toPandas(), expected.toPandas())
-
-
-def test_cast(con):
-    table = con.table("basic_table")
-
-    result = table.mutate(id_string=table.id.cast("string")).compile()
+def test_filter(t, df, filter_fn, expected_fn):
+    result = filter_fn(t).execute().reset_index(drop=True)
+    expected = expected_fn(df).reset_index(drop=True)
+    tm.assert_frame_equal(result, expected)
 
-    df = table.compile()
-    df = df.withColumn("id_string", df.id.cast("string"))
 
-    tm.assert_frame_equal(result.toPandas(), df.toPandas())
+def test_cast(t, df):
+    result = t.mutate(id_string=t.id.cast("string")).execute()
+    df = df.assign(id_string=df.id.astype(str))
+    tm.assert_frame_equal(result, df)
 
 
-def test_alias_after_select(con):
+def test_alias_after_select(t, df):
     # Regression test for issue 2136
-    table = con.table("basic_table")
-    table = table[["id"]]
+    table = t[["id"]]
     table = table.mutate(id2=table["id"])
-
-    result = table.compile().toPandas()
+    result = table.execute()
     tm.assert_series_equal(result["id"], result["id2"], check_names=False)
 
 
-@pytest.mark.parametrize(
-    ("selection_fn", "selection_idx", "expected"),
-    [
-        # selected column id is selections[0], OK to replace since
-        # id == t['id'] (straightforward column projection)
-        (lambda t: t[["id"]], 0, True),
-        # new column v is selections[1], cannot be replaced since it does
-        # not exist in the root table
-        (lambda t: t.mutate(v=t["id"]), 1, False),
-        # new column id is selections[0], cannot be replaced since
-        # new id != t['id']
-        (lambda t: t.mutate(id=t["str_col"]), 0, False),
-        # new column id is selections[0], OK to replace since
-        # new id == t['id'] (mutation is no-op)
-        (lambda t: t.mutate(id=t["id"]), 0, True),
-        # new column id is selections[0], cannot be replaced since
-        # new id != t['id']
-        (lambda t: t.mutate(id=t["id"] + 1), 0, False),
-        # new column id is selections[0], OK to replace since
-        # new id == t['id'] (rename is a no-op)
-        (lambda t: t.rename({"id": "id"}), 0, True),
-        # new column id2 is selections[0], cannot be replaced since
-        # id2 does not exist in the table
-        (lambda t: t.rename({"id2": "id"}), 0, False),
-    ],
-)
-def test_can_be_replaced_by_column_name(selection_fn, selection_idx, expected):
```

Comment:
Guessing this test is no longer useful?

Reply:
Yes, this is rather testing the expression layer.